salt/states/file.py

```diff
@@ -2435,11 +2434,13 @@ def managed(name,
                     'contents_grains is not a string or list of strings, and '
                     'is not binary data. SLS is likely malformed.'
                 )
-            contents = os.linesep.join(
-                [line.rstrip('\n').rstrip('\r') for line in validated_contents]
-            )
+            contents = ''
+            for part in validated_contents:
+                for line in part.splitlines():
+                    contents += line.rstrip('\n').rstrip('\r') + os.linesep
             if contents_newline and not contents.endswith(os.linesep):
                 contents += os.linesep
+            log.error("USE CONTENTS %s %s %s", repr(use_contents), repr(validated_contents), repr(contents))
```

Comment:
Instead of manually calling `repr`, you can use `%r` instead of `%s` and string replacement will automagically call `repr()`.

Reply:
Thanks for pointing that out! This log was for debug so I will remove it.